### PR TITLE
Python linting: Upgrade from pep8 to flake8

### PR DESCRIPTION
**Pull request by cclauss**
As suggested at https://github.com/iovisor/bcc/pull/2431#discussion_r298824303

There are a _ton_ (61) of error codes to eliminate from the __.flake8__ file but many (40) of them could be autocorrected by running the codebase through [__python/black__](https://github.com/python/black).

Tests are failing because of #2445

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by yonghong-song** [comment]
Indeed, looks flake8 does more than pep8 (Flake8 is a tool that inspects code for errors, including failing to comply with PEP8 standards).

**Comment by cclauss** [comment]
Yes.  https://github.com/iovisor/bcc/pull/2431#discussion_r298834881

**Comment by pchaigno** [comment]
Thanks @cclauss!

I'm going to review this in #2438 as it helps see which other errors we might want to ignore.

**Comment by pchaigno** [comment]
Isn't this pull request missing changes to `scripts/py-style-check.sh`?

**Comment by cclauss** [comment]
[buildbot, test this please]  Or does @yonghong-song need to do that?

File: scripts/py-style-check.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@
 set -euo pipefail
 
 # TODO: stop ignoring this. Maybe autopep8 existing stuff?
-find tools -type f -name "*.py" | xargs pep8 -r --show-source --ignore=E123,E125,E126,E127,E128,E302 || \
-    echo "pep8 run failed, please fix it" >&2
+find tools -type f -name "*.py" | xargs flake8 || echo "flake8 run failed, please fix it" >&2
+flake8 . || echo "flake8 run failed, please fix it" >&2
```
Comment (pchaigno) [inline scripts/py-style-check.sh]: The second line makes the first useless.

Could we restrict this to `tools` (and maybe a few other directories)?

Comment (cclauss) [reply scripts/py-style-check.sh]: Yes.  I was experimenting in the vain hope that the CI would actually run.  We are looking for syntax errors.  Probably better that the linter finds them in our examples than our users do.

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by cclauss** [comment]
Is is just me or what?

https://buildbot.iovisor.org/jenkins gives me __502 Bad Gateway nginx/1.10.3 (Ubuntu)__

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Pull request closed without merging**